firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -81,8 +96,15 @@ static CrossProcessLock acquire(Context appContext, String lockName) {
   /** Release a previously acquired lock and free any underlying resources. */
   void releaseAndClose() {
     try {
-      lock.release();
-      channel.close();
+      if (lock != null) {
+        lock.release();
+      }
+      if (channel != null) {
+        channel.close();
+      }
+      if (randomAccessFile != null) {
+        randomAccessFile.close();
```

Comment:
afaict, if any of the above calls throw, the file won't get closed. This is a one more reason to use try-with-resources as doing it manually is prone to bugs

Reply:
Updated as per your suggestion. Another look.